packages/core/core/src/applyRuntimes.js

```diff
@@ -76,8 +79,20 @@ export default async function applyRuntimes({
         if (applied) {
           let runtimeAssets = Array.isArray(applied) ? applied : [applied];
           for (let {code, dependency, filePath, isEntry} of runtimeAssets) {
+            let sourceName = path.join(
+              path.dirname(filePath),
+              `runtime-${md5FromString(code)}.${bundle.type}`,
+            );
+
+            let sourcemap = SourceMap.generateEmptyMap({
+              projectRoot: pluginOptions.projectRoot,
+              sourceName,
+              sourceContent: code,
+            });
+
             let assetGroup = {
               code,
+              mapBuffer: sourcemap.toBuffer(),
```

Comment:
This is still going to be an issue as it (along with the code) gets serialized into the bundlegraph. cc @DeMoorJasper @devongovett

Reply:
Yes. Do we actually need the sourcemap here? It's empty and only contains the same code as is already passed through the request. If there's no existing map, won't it already be set as the source content during transformation?

Reply:
Another issue is that with empty maps all of the columns mappings are zero when the empty map is extended. If we overwrote the `filePath` here though it breaks a lot of tests that expect `JSRuntime.js` as the name. Maybe the test util should just ignore runtime assets?